doc/devel/documenting_mpl.rst

```diff
@@ -29,95 +29,177 @@ requirements that are needed to build the documentation.  They are listed in
+            Respective beginning and end of each line. If scalars are
+            provided, all lines will have same length.
+
+        colors : array_like of colors, optional, default: 'k'
+
+        linestyles : ['solid' | 'dashed' | 'dashdot' | 'dotted'], optional
+
+        label : string, optional, default: ''
+
+        Returns
+        -------
+        lines : `~matplotlib.collections.LineCollection`
+
+        Other Parameters
+        ----------------
+        **kwargs : `~matplotlib.collections.LineCollection` properties.
+
+        See also
+        --------
+        vlines : vertical lines
+        axhline: horizontal line across the axes
+        """
 
-docstrings
-----------
+The Sphinx website also contains plenty of documentation_ concerning ReST
+markup and working with Sphinx in general.
 
-In addition to the "narrative" documentation described above,
-Matplotlib also defines its API reference documentation in docstrings.
-For the most part, these are standard Python docstrings, but
-Matplotlib also includes some features to better support documenting
-getters and setters.
+Linking to other code
+---------------------
+To link to other methods, classes, or modules in Matplotlib you can encase
+the name to refer to in back ticks, for example:
 
+.. code-block:: python
+
+  `~matplotlib.collections.LineCollection`
+
+It is also possible to add links to code in Python, Numpy, Scipy, or Pandas.
+Sometimes it is tricky to get external Sphinx linking to work; to check that
+a something exists to link to the following shell command outputs a list of all
+objects that can be referenced (in this case for Numpy)::
+
+  python -m sphinx.ext.intersphinx 'https://docs.scipy.org/doc/numpy/objects.inv'
+
+
+Function arguments
+------------------
+Function arguments and keywords within docstrings should be referred to using
+the ``*emphasis*`` role. This will keep Matplotlib's documentation consistent
+with Python's documentation:
+
+.. code-block:: rst
+
+  Here is a description of *argument*
+
+Please do not use the ```default role```:
+
+
+.. code-block:: rst
+
+   Please do not describe `argument` like this.
+
+nor the ````literal```` role:
+
+.. code-block:: rst
+
+   Please do not describe ``argument`` like this.
+
+Setters and getters
+-------------------
 Matplotlib uses artist introspection of docstrings to support properties.
 All properties that you want to support through `~.pyplot.setp` and
 `~.pyplot.getp` should have a ``set_property`` and ``get_property`` method in
-the `~.matplotlib.artist.Artist` class.  Yes, this is not ideal given python
-properties or enthought traits, but it is a historical legacy for now.  The
-setter methods use the docstring with the ACCEPTS token to indicate the type of
-argument the method accepts. e.g., in `.Line2D`:
+the `~.matplotlib.artist.Artist` class. The setter methods use the docstring
+with the ACCEPTS token to indicate the type of argument the method accepts.
+e.g., in `.Line2D`:
```

Comment:
OK, I've written some docs, and have used matplotlib for a while.  I don't know what the first sentence of this section means.  Can we clarify for the somewhat uninitiated?

Reply:
Something like
> Artist properties are implemented using setter and getter methods (because Matplotlib predates the introductions of the `property` decorator in Python).  By convention, these setters and getters are named ``set_PROPERTYNAME`` and ``get_PROPERTYNAME``; the list of properties thusly defined on an artist and their values can be listed by the `~.pyplot.setp` and `~.pyplot.getp` functions.

> Property setter methods should indicate the values they accept using a (legacy) special block in the docstring, starting with ``ACCEPTS``, as follows: (code example follows)

> The ACCEPTS block is used to render a table of all properties and their acceptable values in the docs; it can also be displayed using, e.g., ``plt.setp(Line2D)`` (all properties) or ``plt.setp(Line2D, 'linestyle')`` (just one property).

looks good?

Reply:
Yes that makes sense to me.  Thanks!

Reply:
@dstansby can you apply the patch? or I can do it too

Reply:
Feel free to push any changes to this section - I don't have the bandwidth to get my head around it at the moment!